### PR TITLE
ISPN-6946 Redundand indexing during topology changes in the AffinityI…

### PR DESCRIPTION
**Pull request by gustavonalle**
…ndexManager

https://issues.jboss.org/browse/ISPN-6946

### CODE REVIEW

**Comment by anistor** [comment]
Looks good.

File: server/integration/infinispan/src/main/resources/schema/jboss-infinispan-core_9_0.xsd [hunk 1]
```diff
@@ -1569,6 +1569,11 @@
                     <xs:documentation>Index all data</xs:documentation>
                 </xs:annotation>
             </xs:enumeration>
+            <xs:enumeration value="PRIMARY_OWNER">
+                <xs:annotation>
+                    <xs:documentation>Only index changes on the primary owner, regardless of it's local or remote.</xs:documentation>
```
Comment (anistor) [inline server/integration/infinispan/src/main/resources/schema/jboss-infinispan-core_9_0.xsd]: regardless of/ regardless if ?

Comment (gustavonalle) [inline server/integration/infinispan/src/main/resources/schema/jboss-infinispan-core_9_0.xsd]: "regardless of" sounds correct

Comment (anistor) [inline server/integration/infinispan/src/main/resources/schema/jboss-infinispan-core_9_0.xsd]: ok

File: query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java [hunk 2]
```diff
@@ -391,11 +380,11 @@ public void setSearchWorkCreator(SearchWorkCreator<Object> searchWorkCreator) {
     * @param transactionContext Optional for lazy initialization, or reuse an existing context.
     */
    private void processReplaceCommand(final ReplaceCommand command, final InvocationContext ctx, final Object valueReplaced, TransactionContext transactionContext) {
-      if (valueReplaced != null && command.isSuccessful() && shouldModifyIndexes(command, ctx)) {
+      Object key = extractValue(command.getKey());
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java]: I'd move extractValue inside the ```if```

File: query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java [hunk 3]
```diff
@@ -421,11 +410,12 @@ private void processReplaceCommand(final ReplaceCommand command, final Invocatio
     * @param transactionContext Optional for lazy initialization, or reuse an existing context.
     */
    private void processRemoveCommand(final RemoveCommand command, final InvocationContext ctx, final Object valueRemoved, TransactionContext transactionContext) {
-      if (command.isSuccessful() && !command.isNonExistent() && shouldModifyIndexes(command, ctx)) {
+      Object key = extractValue(command.getKey());
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java]: Same as above.

File: query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java (hunk 2 above)
Comment (gustavonalle) [inline query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java]: Unfortunately to decide if ```shouldModifyIndexes``` the ```key``` is needed ahead of time

Comment (anistor) [inline query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java]: The ```if``` can be split in two

Comment (gustavonalle) [inline query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java]: np, will change it

**Comment by gustavonalle** [comment]
@anistor updated

**Comment by anistor** [comment]
Pulling...

**Comment by anistor** [comment]
Integrated. Thanks @gustavonalle !

**Pull request closed without merging**